From e33d096de11736d03b53ef2f0b3276e239cf60b1 Mon Sep 17 00:00:00 2001
From: Athaariq Ardhiansyah <buildroot@athaariq.my.id>
Date: Sat, 1 Oct 2022 02:05:19 +0000
Subject: [PATCH 5/7] package/glibc: fix fsconfig_command redeclaration bug

Temporary fix for buildroot-specific bug. Ideally, we should install linux-headers first then glibc. Is this actually possible on Buildroot? If yes, then we can also fix bug #15021 simultaneously if linux-headers being installed first as possible.

Signed-off-by: Athaariq Ardhiansyah <buildroot@athaariq.my.id>
---
 .../0002-fsconfig_command-workaround.patch      | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)
 create mode 100644 package/glibc/0002-fsconfig_command-workaround.patch

diff --git a/package/glibc/0002-fsconfig_command-workaround.patch b/package/glibc/0002-fsconfig_command-workaround.patch
new file mode 100644
index 0000000000..27dd25b244
--- /dev/null
+++ b/package/glibc/0002-fsconfig_command-workaround.patch
@@ -0,0 +1,17 @@
+diff --git a/sysdeps/unix/sysv/linux/sys/mount.h b/sysdeps/unix/sysv/linux/sys/mount.h
+index 19841d0738..43a681d097 100644
+--- a/sysdeps/unix/sysv/linux/sys/mount.h
++++ b/sysdeps/unix/sysv/linux/sys/mount.h
+@@ -27,11 +27,7 @@
+ #include <stddef.h>
+ #include <sys/ioctl.h>
+ 
+-#ifdef __has_include
+-# if __has_include ("linux/mount.h")
+-#  include "linux/mount.h"
+-# endif
+-#endif
++#include "linux/mount.h"
+ 
+ 
+ #define BLOCK_SIZE_BITS	10
-- 
2.37.3

